seek_to.h

```diff
@@ -0,0 +1,25 @@
+
+#include "leveldb/c.h"
+#include <stdbool.h>
+
+typedef struct {
+    unsigned char valid;
+    unsigned char equal;
+} SeekResult;
+
+SeekResult leveldb_iter_seek_to(leveldb_iterator_t* iter, const char* k, size_t klen) {
```

Comment:
code in a .h file is kinda poor form due to the way the preprocessor works;  check out https://github.com/DataDog/czlib/blob/master/fastzlib.h and https://github.com/DataDog/czlib/blob/master/fastzlib.c I think it's pretty straightforward to give the implementation a .c file and have things JustWork.